### PR TITLE
feat(bot): renovate auto merge support

### PR DESCRIPTION
**Pull request by chdsbd**
This change supports auto merging Renovate PRs by update type.

related #701

### CODE REVIEW

File: bot/kodiak/dependencies.py [hunk 1]
```diff
@@ -78,10 +78,59 @@ def dep_version_from_title(x: str) -> MatchType | None:
...
+    largest_match_type: MatchType | None = None
+    if "| lockFileMaintenance |" in body:
+        largest_match_type = "patch"
+    for match in renovate_body_regex.finditer(body):
+        group = match.groupdict()
+        if "old_version" not in group or "new_version" not in group:
+            continue
+        match_type = _compare_versions(group["old_version"], group["new_version"])
+        if not match_type:
+            continue
```
Comment (sbdchd) [inline bot/kodiak/dependencies.py:120]: what about these codecov warnings?

Comment (chdsbd) [reply bot/kodiak/dependencies.py:120]: fixed

File: bot/kodiak/tests/dependencies/test_dependencies.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from kodiak.dependencies import dep_versions_from_pr
+
+
+@dataclass
+class FakePR:
+    title: str
+    body: str
+
+
+def generate_test_cases() -> Iterator[tuple[FakePR, str]]:
```
Comment (sbdchd) [inline bot/kodiak/tests/dependencies/test_dependencies.py:19]: very fancy

File: bot/kodiak/tests/dependencies/test_dependencies.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
```
Comment (sbdchd) [inline bot/kodiak/tests/dependencies/test_dependencies.py:2]: should this test module be renamed to say "test_renovate.py` like the dependabot one?

Comment (chdsbd) [reply bot/kodiak/tests/dependencies/test_dependencies.py:2]: no, because this is general, testing both of them

**Review by sbdchd — APPROVED** [review]

**Pull request merged**